currency_rate_update/model/currency_rate_update.py

```diff
@@ -243,34 +129,24 @@ def refresh_currency(self):
         _logger.info(
             'Starting to refresh currencies with service %s (company: %s)',
             self.service, self.company_id.name)
-        factory = Currency_getter_factory()
-        curr_obj = self.env['res.currency']
         rate_obj = self.env['res.currency.rate']
         company = self.company_id
         # The multi company currency can be set or no so we handle
         # The two case
         if company.auto_currency_up:
-            main_currency = curr_obj.search(
-                [('base', '=', True), ('company_id', '=', company.id)],
-                limit=1)
-            if not main_currency:
-                # If we can not find a base currency for this company
-                # we look for one with no company set
-                main_currency = curr_obj.search(
-                    [('base', '=', True), ('company_id', '=', False)],
-                    limit=1)
+            main_currency = self.company_id.currency_id
             if not main_currency:
-                raise exceptions.Warning(_('There is no base currency set!'))
+                raise exceptions.Warning(_('There is no main \
```

Comment:
Please separate the string that way:

_('There is no main '
   'currency defined!')

Here I think you are introducting many spaces between main and currency